src/main/scala/com/mongodb/spark/rdd/partitioner/MongoPaginateByCountPartitioner.scala

```diff
@@ -63,25 +63,28 @@ class MongoPaginateByCountPartitioner extends MongoPartitioner with MongoPaginat
    * @return the partitions
    */
   override def partitions(connector: MongoConnector, readConfig: ReadConfig, pipeline: Array[BsonDocument]): Array[MongoPartition] = {
-
-    Try(PartitionerHelper.collStats(connector, readConfig)) match {
-      case Success(results) =>
+    val matchQuery = PartitionerHelper.matchQuery(pipeline)
```

Comment:
This partitioner now uses a `count` query with the `matchQuery` to get the size of the results rather than `collstats`.

If there is a `$match` filter the last partition is changed to be inclusive of the last value by using `setLastBoundaryToLessThanOrEqualTo`.